lib/git_worktree.rb

```diff
@@ -167,6 +167,19 @@ def file_list
     tree.walk(:preorder).collect { |root, entry| "#{root}#{entry[:name]}" }
   end
 
+  # Like "file_list", but doesn't return directories
+  def blob_list &block
+    tree = lookup_commit_tree
+    if block_given?
+      tree.walk_blobs(:preorder, &block)
```

Comment:
Is this leg of the conditional used?  Curious why this was added?

Reply:
So I started off with just the lower leg to handle the "nested playbooks" case, which in one of the earlier commits, was able to just be a file extension check, but was very similar to `file_list` in implementation.

But once I needed to use the `entry` from this to make use of `read_entry` so I could validate contents, then the block form was needed.  I left the previous form so it could be used "like `file_list`" if desired.  `.walk_blobs` unfortunately requires a block, and isn't like `.walk` which will return an enumerator if one isn't passed.

Reply:
I'm not sure what walk_blobs is getting you if you are using read_entry anyway.  That is, you can also just use file_list and then use read_file (which uses read_entry under the covers).

Either way, read_file or read_entry, you get the entire content back, and once that happens it's already in memory.  At first I was thinking somehow walk_blobs returned something special that prevented the entire blob from being in memory at once, but it doesn't.

Reply:
Left a big "blobby" comment answering this here:

https://github.com/ManageIQ/manageiq/pull/19089#discussion_r309936250

Reply:
Didn't rework `.file_list` to work like `.blob_list` since I didn't want to be that invasive in this PR, but I did remove a decent amount of the logic in the rebase, so it should be more around what you requested.